v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/V2FUtils.scala

```diff
@@ -100,44 +77,25 @@ object V2FUtils {
       }
     }
 
-  /**
-    * Converts a string in camel case format to a string in snake case format using regex, replaces - with _'s, and puts underscores before and after every number.
-    *
-    * @param string the string being converted
-    */
-  def camelCaseToSnakeCase(string: String): String = {
-    string
-      .replace(
-        "-",
-        "_"
-      )
-      .replaceAll(
-        "([A-Z]+)([A-Z][a-z])",
-        "$1_$2"
-      )
-      .replaceAll("([a-z\\d])([A-Z])", "$1_$2")
-      .replaceAll("([a-z])([\\d])", "$1_$2")
-      .replaceAll("([\\d])([a-z])", "$1_$2")
-      .toLowerCase
-  }
-
   // the pattern of where the ancestryID is located in the tsv path
   // e.g: gs://path/to/metaanalysis/ancestry-specific/phenotype/ancestry=ancestryID/file
   val ancestryIDPattern: Regex = "\\/ancestry=([^\\/]+)\\/".r
 
   /**
-    * Adds the ancestry ID from the ReadableFile paths as a field with a key, value of "ancestry" -> ancestryID to the JSON Objects.
+    * Adds the ancestry ID from the ReadableFile paths as a field with a key, value of "ancestry" -> ancestryID to the Msg Objects.
     *
-    * @param tableName the name of the tsv table that was converted to json
+    * @param tableName the name of the tsv table that was converted to Msg
     */
   def addAncestryID(
     tableName: String
-  ): SCollection[(String, JsonObject)] => SCollection[(String, JsonObject)] = {
-    _.transform(s"Adding the ancestry ID from $tableName's tsv path to its json") {
+  ): SCollection[(String, Msg)] => SCollection[(String, Msg)] = {
+    _.transform(s"Adding the ancestry ID from $tableName's tsv path to its Msg object") {
       collection =>
         collection.map {
-          case (filePath, jsonObj) =>
-            val ancestryID = Json.fromString(
+          case (filePath, msgObj) =>
+            val toRet = upack.copy(msgObj)
+            val underlying = toRet.obj
+            val ancestryID = Str(
```

Comment:
Nitpick: I think it'd be better style to have the `Str` wrapper around `ancestryID` in the call to `update` below, instead of here. That shows more clearly that the `Str` is only there to make the update type-check.